core/dive.c

```diff
@@ -1999,43 +2029,54 @@ static int same_gas(struct event *a, struct event *b)
 	return false;
 }
 
-static void merge_events(struct divecomputer *res, struct divecomputer *src1, struct divecomputer *src2, int offset)
+static void event_renumber(struct event *ev, const int mapping[]);
+static void add_initial_gaschange(struct dive *dive, struct divecomputer *dc, int offset, int idx);
+
+static void merge_events(struct dive *d, struct divecomputer *res,
+			 const struct divecomputer *src1, const struct divecomputer *src2,
+			 const int *cylinders_map1, const int *cylinders_map2,
+			 int offset)
 {
-	struct event *a, *b;
+	const struct event *a, *b;
 	struct event **p = &res->events;
-	struct event *last_gas = NULL;
+	const struct event *last_gas = NULL;
 
 	/* Always use positive offsets */
 	if (offset < 0) {
-		struct divecomputer *tmp;
+		const struct divecomputer *tmp;
+		const int *cylinders_map_tmp;
 
 		offset = -offset;
 		tmp = src1;
 		src1 = src2;
 		src2 = tmp;
+
+		cylinders_map_tmp = cylinders_map1;
+		cylinders_map1 = cylinders_map2;
+		cylinders_map2 = cylinders_map_tmp;
 	}
 
 	a = src1->events;
 	b = src2->events;
-	while (b) {
-		b->time.seconds += offset;
-		b = b->next;
-	}
-	b = src2->events;
```

Comment:
so you remove the event adjustment here because you now do consider the offset as part of the sort... but I'm not quite understanding why you are making that change

Reply:
The whole purpose of this commit is to not modify the original dive data. So we can't adjust the offset of the events of the `src2` dive. :(